hydra-plutus/src/Hydra/Contract/Util.hs

```diff
@@ -20,13 +20,14 @@ hydraHeadV1 = "HydraHeadV1"
 -- 'CurrencySymbol' and 'TokenName' of 'hydraHeadV1'
 hasST :: CurrencySymbol -> Value -> Bool
 hasST headPolicyId v =
-  isJust $
-    find
-      (\(cs, tokenMap) -> cs == headPolicyId && hasHydraToken tokenMap)
-      (Map.toList $ getValue v)
+  case hasST' of
+    Just True -> True
+    _otherwise -> traceError "ST not spent in transaction"
```

Comment:
`hasST` does not contain any info on "spending" or not. `Value` could be the minted value of  a transaction. Also, we have a `Bool` result, so let's use it.

Should: Not include an error but just use the result of the `Maybe` monad. e.g. `maybe False id $ do ... `

Reply:
@ch1bo see new version